app/models/service_request.rb

```diff
@@ -598,6 +598,16 @@ def has_non_first_draft_ssrs?
     sub_service_requests.where.not(status: 'first_draft').any?
   end
 
+  def ssrs_associated_with_service_provider (service_provider)
```

Comment:
Could be a cleaner solution, but honestly it is just a different way of doing it. Fine if we keep it how it is.